docs/resources/workspace_service.md

```diff
@@ -0,0 +1,198 @@
+---
+subcategory: "Workspace"
+---
+
+# huaweicloud_workspace_service
+
+Use this resource to register or unregister the Workspace service in HuaweiCloud.
+
+## Example Usage
+
+### Register the Workspace service and use local authentication
+
+```hcl
+variable "vpc_id" {}
+variable "network_ids" {}
+
+resource "huaweicloud_workspace_service" "test" {
+  access_mode = "INTERNET"
+  vpc_id      = var.vpc_id
+  network_ids = var.network_ids
+}
+```
+
+### Register the Workspace service and connect to the AD domain
+
+```hcl
+variable "vpc_id" {}
+variable "network_ids" {}
+
+variable "ad_domain_name" {}
+variable "ad_server_admin_account" {}
+variable "ad_server_admin_password" {}
+variable "ad_master_domain_ip" {}
+variable "ad_server_name" {}
+variable "ad_master_dns_ip" {}
+
+resource "huaweicloud_workspace_service" "test" {
+  auth_type   = "LOCAL_AD"
+  access_mode = "INTERNET"
+  vpc_id      = var.vpc_id
+  network_ids = var.network_ids
+
+  ad_domain {
+    name               = var.ad_domain_name
+    admin_account      = var.ad_server_admin_account
+    password           = var.ad_server_admin_password
+    active_domain_ip   = var.ad_master_domain_ip
+    active_domain_name = format("%s.%s", var.ad_server_name, var.ad_domain_name)
+    active_dns_ip      = var.ad_master_dns_ip
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to register the Workspace service.
+  If omitted, the provider-level region will be used. Changing this will create a new resource.
+
+* `vpc_id` - (Required, String) Specifies the VPC ID to which the service belongs.
+
+  -> The resources required by Workspace will be created in the selected VPC subnet. After the configuration is saved,
+     the VPC cannot be modified.
+
+* `network_ids` - (Required, List) The network ID list of subnets that the service have.
+  The subnets corresponding to this parameter must be included in the VPC resource corresponding to `vpc_id`.
+  These subnet segments cannot conflict with `172.16.0.0/12`.
+
+  -> The subnet of first registry must be selected. The DNS server address of the selected subnet will be automatically
+     changed. Do not manually change it. You are advised to select a dedicated Workspace subnet and ensure that the DHCP
+     function of the subnet is enabled.
+
+* `access_mode` - (Required, String) Specifies the access mode of Workspace service.
+  The valid values are as follows:
+  + **INTERNET**: internet access.
+  + **DEDICATED**: dedicated line access.
+  + **BOTH**: both internet access and dedicated access are supported.
+
+* `auth_type` - (Optional, String, ForceNew) Specifies the authentication type of Workspace service.
```

Comment:
It's not ForceNew in the code

Reply:
The `ForceNew` behavior is missing in the schema.